packages/generator/src/edmx-to-vdm/common/complex-type.ts

```diff
@@ -65,11 +63,23 @@ export function transformComplexTypesBase(
           c.Name,
           p.Name
         );
-        const isComplex = isComplexType(p.Type);
         const isCollection = isCollectionType(p.Type);
+        const isComplexOrEnum = isComplexTypeOrEnumType(
```

Comment:
goes also in the direction of my comment above. I find this very complicated, which is not avoidable since there are so many cases. But I would do the following. 
- Put all the isSomething methods in the edmx-to-vdm-util class.
- Include the collection check inside the isSomething method. So isEnum() returns true on a collection of enums and a enum input. Since it is used this way anyway it will reduce complexity in the coden
- I know it is more efficient the way you done it. For example if `isComplexOrEnum=true` and you later find isEnum=true it can not be a complex. However, I would find a less efficient but more symmetric flow nice:
```
isCollection = isCollection(string)
isEnum= isEnum(string,enums) //uses the enumOrComplex internally
isComplex = isComplex(string,complexTypess) //uses the enumOrComplex internally
```